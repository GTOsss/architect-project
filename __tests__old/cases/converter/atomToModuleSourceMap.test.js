const stringifyObject = require('stringify-object');
const { sourceMapToModule } = require('../../../src/utils/change/atomToModuleSourceMap');
const { aliases, defaultParams, map } = require('./source_map/source-map-atom');

const createPrettyObj = (obj) => {
  return stringifyObject(obj, {
    indent: '  ',
    singleQuotes: true,
  });
};

describe('test atomToModule', () => {
  test('simple with atom defaultParams', () => {
    const obj = sourceMapToModule({
      map,
      defaultParams,
      aliases: {},
    });

    expect(createPrettyObj(obj)).toMatchSnapshot();
  });

  test('simple with aliases', () => {
    const obj = sourceMapToModule({
      map,
      defaultParams: {},
      aliases,
    });

    expect(createPrettyObj(obj)).toMatchSnapshot();
  });
});
