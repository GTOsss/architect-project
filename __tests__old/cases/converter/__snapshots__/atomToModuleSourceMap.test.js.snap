// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test atomToModule simple with aliases 1`] = `
"{
  map: {
    src: {
      stadiums: 'store'
    },
    'src/components/pages': {
      stadiums: 'component'
    }
  },
  aliases: {
    c: 'component',
    p: 'page',
    s: 'store',
    ix: 'index'
  }
}"
`;

exports[`test atomToModule simple with atom defaultParams 1`] = `
"{
  map: {
    'src/pages': {
      stadiums: 'page'
    },
    'src/components/pages': {
      stadiums: 'component'
    },
    'src/store': {
      stadiums: 'store'
    }
  },
  aliases: {}
}"
`;
